corehq/util/metrics/__init__.py

```diff
@@ -0,0 +1,43 @@
+from typing import Iterable
+
+import settings
+from corehq.util.metrics.metrics import DummyMetrics, DelegatedMetrics, DEFAULT_BUCKETS
+from dimagi.utils.modules import to_function
+
+_metrics = None
+
+
+def _get_metrics_provider():
+    global _metrics
+    if not _metrics:
+        providers = []
+        for provider_path in settings.METRICS_PROVIDERS:
+            provider = to_function(provider_path)()
+            provider.initialize()
+            providers.append(provider)
+
+        if not providers:
+            _metrics = DummyMetrics()
+        elif len(providers) > 1:
+            _metrics = DelegatedMetrics(providers)
+        else:
+            _metrics = providers[0]
```

Comment:
Any reason not to always use `DelegatedMetrics` here, with sometimes only one provider, possibly only `DummyMetrics`?

Reply:
`DelegatedMetrics` adds extra/unnecessary complexity when there is only one provider. It's an extra layer of indirection and adds (small) runtime overhead for every metric event, which is avoided entirely here. The new metrics provider abstraction also adds more indirection than we had before.